frontend/src/utils/index.ts

```diff
@@ -30,15 +29,20 @@ export const b64toBlob = base64 => fetch(base64).then(res => res.blob())
 
 /**
  * Get the repo name from its url
+ * !if `url` is invalid the form validation would have caught it
  * @param url
  * @returns {string}
  * @example
  * // returns 'ulx3s'
  * urlToName('https://github.com/emard/ulx3s/')
  */
 export const urlToName = url => {
-  const urlObj = new URL(url)
-  return path.basename(urlObj.pathname, path.extname(urlObj.pathname))
+  try {
+    url = new URL(url)
+  } catch {
+    return ''
+  }
+  return path.basename(url.pathname, path.extname(url.pathname))
```

Comment:
I thought you changed this while we were pair programming?

Reply:
The cleanest way to check if a string is a valid URL is to construct a URL object.

Checking if the value is falsey or empty string won't work, because a non-empty string would pass that check and cause an exception when calling `new URL(val)`.
We didn't face this problem because we were copy-pasting the URL, so we never called this function with a string which is invalid URL.